src/blockfilter.cpp

```diff
@@ -248,7 +248,8 @@ static GCSFilter::ElementSet BasicFilterElements(const CBlock& block,
     for (const CTransactionRef& tx : block.vtx) {
         for (const CTxOut& txout : tx->vout) {
             const CScript& script = txout.scriptPubKey;
-            if (script.empty() || script[0] == OP_RETURN) continue;
+            if (script.empty()) continue;
+            if (script[0] == OP_RETURN && script.IsPushOnly(script.begin() + 1)) continue;
```

Comment:
This change doesn't appear consistent with BIP158. ("The scriptPubKey of each output, aside from all OP_RETURN output scripts.", "We exclude all OP_RETURN outputs in order to allow filters to easily be committed to in the future via a soft-fork.")  Also I don't think it's a good idea? They're no less unspendable even if they're not push only.

Reply:
Yeah, the BIP should probably clarify exactly what that means. I made this change because 1) btcd does it and 2) the standard TX_NULL_DATA script type is [defined this way](https://github.com/bitcoin/bitcoin/blob/b68bac83d9ead9e8893785a58fbed6d09cba4b5c/src/script/standard.cpp#L128). In other words, scripts beginning with OP_RETURN and not followed by pushdata are non-standard. I think it makes sense to stick with this definition of "OP_RETURN output", but don't feel too strongly.

Reply:
We should probably match IsUnspendable (ie that it just starts with OP_RETURN). TX_NULL_DATA is just an internal thing, not something to mirror this based on.

Reply:
I agree with @TheBlueMatt that there is no need to leak our internal TX_NULL_DATA template into the BIP. Would it make sense to use CScript::IsUnspendable instead (or at least equivalent to its current definition), which triggers on *((length > 0 and first_byte = OP_RETURN) or length > 10000)*.

Reply:
This has been clarified in the BIP and updated here.